tests/publiccloud/upload_container_images.pm

```diff
@@ -0,0 +1,85 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Push a container image to the tested public cloud container registry
+#
+# Maintainer: Ivan Lausuch <ilausuch@suse.com>, qa-c team <qa-c@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use publiccloud::ecr;
+use containers::urls 'get_suse_container_urls';
+use containers::common qw(install_docker_when_needed allow_selected_insecure_registries);
+use version_utils qw(is_sle get_os_release);
+
+sub run {
+    my ($self, $args) = @_;
+    $self->select_serial_terminal;
+
+    my ($running_version, $sp, $host_distri) = get_os_release;
+    install_docker_when_needed($host_distri);
+    allow_selected_insecure_registries(runtime => 'docker') if (is_sle());
+
+    my $provider = publiccloud::ecr->new(
+        key_id     => get_var('PUBLIC_CLOUD_KEY_ID'),
+        key_secret => get_var('PUBLIC_CLOUD_KEY_SECRET'),
+        region     => get_var('PUBLIC_CLOUD_REGION', 'eu-central-1'),
+        username   => get_var('PUBLIC_CLOUD_USER',   'ec2-user')
+    );
+    $provider->init();
```

Comment:
I tested that the init() is necessary because this provider is not the same created by the basetest. And as the basetest it needs to be initialized, check:

https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/4be33ca50080a9bc942e02299ad52cde8f18ab5c/lib/publiccloud/basetest.pm#L65